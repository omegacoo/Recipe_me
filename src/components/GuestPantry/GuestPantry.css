/* Mobile Phone Portrait */
.GuestPantry_ {
    box-shadow: 0 -5px 10px -5px;
}

.GuestPantry_title {
    margin: 4vh;
    margin-bottom: 0vh;
    margin-top: 2vh;
    text-align: center;
    font-size: 10vw;
}

.GuestPantry_icons {
    width: fit-content;
    margin-left: auto;
    margin-right: auto;
    display: grid;
    grid-template-columns: 50% 50%;
    grid-template-rows: 18vh 18vh 18vh;
}

.GuestPantry_icons img {
    border-bottom: .5vh solid rgb(90, 88, 80);
    border-right: .5vh solid rgb(90, 88, 80);
    border-radius: 15px;
    margin: 2vh;
    padding-top: 1px;
    background-color: #F0EAD6;
    overflow: hidden;
    width: 17vh;
    box-shadow: 10px 10px 10px black;
}

.GuestPantry .GuestPantry_popupContainer {
    position: absolute;
}

.GuestPantry_popup {
    z-index: 1;
    background-image: url('../../assets/pad.png');
    background-size: 100% 100%;
    background-repeat: no-repeat;
    background-color: transparent;
    color: black;
    padding: 2vh;
    position: absolute;
    width: min-content;
    font-weight: bold;
    width: 90vw;
    height: 70vh;
    top: -60vh;
    left: 10vw;
    text-align: center;
}

.GuestPantry_popup .GuestPantry_closePopup {
    background-color: red;
    color: #F0EAD6;
    border-radius: 50%;
    font-size: 3vh;
    position: absolute;
    left: 7vw;
    top: 5vh;
    box-shadow: -3px 0px 3px black;
}

.GuestPantry_list {
    height: 65%;
    width: 70%;
    overflow: auto;
    background-color: transparent;
    border: none;
    margin-left: 12%;
    margin-right: auto;
    text-align: left;
    list-style: none;
}

.GuestPantry_list li {
    margin: 1vh;
    margin-top: 3vh;
    height: fit-content;
    line-height: 2.3vh;
}

.GuestPantry_list li input {
    float: left;
    top: -1vh;
    margin-right: 2vw;
}

.GuestPantry_list li label {
    font-family: 'Indie Flower', cursive;
    font-size: 3vh;
}

.GuestPantry_checkbox {
    position: relative;
    top: 1vw;
    appearance: none;
    width: 6vw;
    height: 6vw;
}

.GuestPantry_checkbox:checked {
    content: url('../../assets/check.png');
}

.GuestPantry_popup_title {
    margin-top: 7vh;
    font-size: 3vh;
    text-shadow: 0 -1px 0 #000;
    border-bottom: 1px solid black;
    width: 30%;
    margin-left: 29%;
    margin-right: auto;
}

.GuestPantry_recipes {
    padding: 5px;
    display: block;
    margin-top: 4vh;
    margin-left: auto;
    margin-right: auto;
    font-weight: bolder;
    width: 30vw;
}

.arrow_one {
    width: 30%;
    position: absolute;
    top: 46%;
}

.arrow_two {
    width: 30%;
    position: absolute;
    top: 28%;
    z-index: 3;
}

.arrow_three {
    width: 30%;
    position: absolute;
    top: 33%;
    z-index: 3;
}

.arrow_four {
    width: 30%;
    position: absolute;
    top: 38%;
    z-index: 3;
}

.arrow_five {
    width: 30%;
    position: absolute;
    top: 17%;
    left: 18%;
    z-index: 3;
}

.arrow_six {
    width: 30%;
    position: absolute;
    top: 46%;
    left: 40%;
    z-index: 3;
}

.arrow_seven {
    width: 30%;
    position: absolute;
    top: 28%;
    z-index: 3;
}

.arrow_eight {
    width: 30%;
    position: absolute;
    top: 17%;
    left: 18%;
    z-index: 3;
}

.arrow_nine {
    width: 30%;
    position: absolute;
    top: 63%;
    left: 40%;
}

.arrow_ten {
    width: 30%;
    position: absolute;
    top: 28%;
    z-index: 3;
}

.arrow_eleven {
    width: 30%;
    position: absolute;
    top: 33%;
    z-index: 3;
}

.arrow_twelve {
    width: 30%;
    position: absolute;
    top: 17%;
    left: 18%;
    z-index: 3;
}

.arrow_thirteen {
    width: 30%;
    position: absolute;
    top: 81%;
    left: 20%;
    z-index: 3;
}

.pulse {
    animation-name: pulse_animation;
    animation-duration: 2000ms;
    transform-origin: 70% 70%;
    animation-iteration-count: infinite;
    animation-timing-function: linear;
    pointer-events: none;
}

@keyframes pulse_animation {
    from { transform: scale(.5) }
    50% { transform: scale(1) }
    to { transform: scale(.5) }
}

/* iPhone X 375x812 Portrait */
@media only screen and (min-width: 375px) and (min-height: 812px) {
    .GuestPantry_title {
        margin-top: 5vh;
    }
}

/* iPad 768x1024 Portrait */
@media only screen and (min-width: 768px) and (min-height: 1024px) {
    .GuestPantry_title {
        text-align: center;
        margin-top: 4vh;
        margin-bottom: -2vh;
    }
    
    .GuestPantry_icons {
        grid-template-rows: 20vh 20vh 20vh;
    }
    
    .GuestPantry_icons img {
        width: 16vh;
    }

    .GuestPantry_recipes {
        width: 15vh;
        margin-top: -1vh;
    }

    .GuestPantry_popup {
        margin-top: -8vh;
    }

    .GuestPantry_popup li {
        font-size: 2.5vh;
    }

    .GuestPantry_popup .closePopup {
        width: fit-content;
        top: 5vh;
    }

    .arrow_one {
        width: 30%;
        position: absolute;
        top: 47%;
        left: 5%
    }
    
    .arrow_two {
        width: 30%;
        position: absolute;
        top: 27%;
        z-index: 3;
    }
    
    .arrow_three {
        width: 30%;
        position: absolute;
        top: 32%;
        z-index: 3;
    }
    
    .arrow_four {
        width: 30%;
        position: absolute;
        top: 37%;
        z-index: 3;
    }
    
    .arrow_five {
        width: 30%;
        position: absolute;
        top: 17%;
        left: 18%;
        z-index: 3;
    }
    
    .arrow_six {
        width: 30%;
        position: absolute;
        top: 48%;
        left: 35%;
        z-index: 3;
    }
    
    .arrow_seven {
        width: 30%;
        position: absolute;
        top: 27%;
        z-index: 3;
    }
    
    .arrow_eight {
        width: 30%;
        position: absolute;
        top: 17%;
        left: 18%;
        z-index: 3;
    }
    
    .arrow_nine {
        width: 30%;
        position: absolute;
        top: 68%;
        left: 35%;
    }
    
    .arrow_ten {
        width: 30%;
        position: absolute;
        top: 27%;
        z-index: 3;
    }
    
    .arrow_eleven {
        width: 30%;
        position: absolute;
        top: 32%;
        z-index: 3;
    }
    
    .arrow_twelve {
        width: 30%;
        position: absolute;
        top: 17%;
        left: 18%;
        z-index: 3;
    }
    
    .arrow_thirteen {
        width: 30%;
        position: absolute;
        top: 82%;
        left: 20%;
        z-index: 3;
    }
}

/* Desktop */
@media screen and (min-width: 1025px) {
    .GuestPantry_title {
        font-size: 5vw;
    }

    .GuestPantry_icons {
        grid-template-columns: 33% 33% 33%;
        grid-template-rows: 18vh 18vh;
        margin-top: 5vh;
    }

    .GuestPantry_popup {
        width: 35vw;
        left: 35vw;
        top: -50vh;
    }

    .GuestPantry_popup .GuestPantry_closePopup {
        left: 3vw;
    }

    .GuestPantry_checkbox {
        width: 2vw;
        height: 2vw;
    }

    .GuestPantry_popup_title {
        margin-top: 8vh;
    }

    .GuestPantry_recipes {
        width: 15vw;
    }

    .GuestPantry_icons img:hover,
    .GuestPantry_checkbox:hover,
    .GuestPantry_popup li:hover,
    .GuestPantry_list li label:hover,
    .GuestPantry_popup .GuestPantry_closePopup:hover,
    .GuestPantry_recipes:hover {
        cursor: pointer;
    }
    
    .GuestPantry_closePopup:active {
        box-shadow: none;
        left: 2.3vw;
    }
    
    .GuestPantry_icons img:active {
        width: 16vh;
        margin: 3vh;
    }
    
    .GuestPantry_recipes:active {
        width: 14vw;
        margin-top: 5vh;
    }

    .arrow_one {
        width: 15%;
        position: absolute;
        top: 28%;
        left: 45%
    }
    
    .arrow_two {
        width: 15%;
        position: absolute;
        top: 21%;
        left: 29%;
        z-index: 3;
    }
    
    .arrow_three {
        width: 15%;
        position: absolute;
        top: 26%;
        left: 29%;
        z-index: 3;
    }
    
    .arrow_four {
        width: 15%;
        position: absolute;
        top: 32%;
        left: 29%;
        z-index: 3;
    }
    
    .arrow_five {
        width: 15%;
        position: absolute;
        top: 10%;
        left: 38%;
        z-index: 3;
    }
    
    .arrow_six {
        width: 15%;
        position: absolute;
        top: 45%;
        left: 23%;
        z-index: 3;
    }
    
    .arrow_seven {
        width: 15%;
        position: absolute;
        top: 21%;
        left: 29%;
        z-index: 3;
    }
    
    .arrow_eight {
        width: 15%;
        position: absolute;
        top: 10%;
        left: 38%;
        z-index: 3;
    }
    
    .arrow_nine {
        width: 15%;
        position: absolute;
        top: 45%;
        left: 48%;
    }
    
    .arrow_ten {
        width: 15%;
        position: absolute;
        top: 21%;
        left: 29%;
        z-index: 3;
    }
    
    .arrow_eleven {
        width: 15%;
        position: absolute;
        top: 26%;
        left: 29%;
        z-index: 3;
    }
    
    .arrow_twelve {
        width: 15%;
        position: absolute;
        top: 10%;
        left: 38%;
        z-index: 3;
    }
    
    .arrow_thirteen {
        width: 15%;
        position: absolute;
        top: 65%;
        left: 35%;
        z-index: 3;
    }
}