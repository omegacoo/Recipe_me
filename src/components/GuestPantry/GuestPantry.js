import React from 'react';

import StoreContext from '../../StoreContext';
import sampleData from '../../sampleData';

import meat from '../../assets/sausage.png';
import vegetable from '../../assets/broccoli.png';
import grain from '../../assets/loaf.png';
import dairy from '../../assets/milk.png';
import condiment from '../../assets/Salt_n_Pepper.png';
import fruit from '../../assets/strawberry.png';
import book from '../../assets/book.png';
import arrow from '../../assets/arrow.png';
import arrowAlt from '../../assets/arrowAlt.png';

import './GuestPantry.css';

export default class GuestPantry extends React.Component {
    static contextType = StoreContext;

    state = {
        currentList: '',
        listOpen: false,
        baseIngredients: this.context.guestIngredients,
        userIngredients: this.context.guestUserIngredients,
        guideStep: 1
    };

    renderGuide = () => {
        let element;

        switch (this.state.guideStep) {
            case 1:
                element = <img src={arrow} alt='arrow' className='arrow_one pulse' />
                break;
            case 2:
                element = <img src={arrow} alt='arrow' className='arrow_two pulse' />
                break;
            case 3:
                element = <img src={arrow} alt='arrow' className='arrow_three pulse' />
                break;  
            case 4:
                element = <img src={arrow} alt='arrow' className='arrow_four pulse' />
                break; 
            case 5:
                element = <img src={arrowAlt} alt='arrow' className='arrow_five pulse' />
                break;
            case 6:
                element = <img src={arrow} alt='arrow' className='arrow_six pulse' />
                break;
            case 7:
                element = <img src={arrow} alt='arrow' className='arrow_seven pulse' />
                break;
            case 8:
                element = <img src={arrowAlt} alt='arrow' className='arrow_eight pulse' />
                break;
            case 9:
                element = <img src={arrow} alt='arrow' className='arrow_nine pulse' />
                break;
            case 10:
                element = <img src={arrow} alt='arrow' className='arrow_ten pulse' />
                break;
            case 11:
                element = <img src={arrow} alt='arrow' className='arrow_eleven pulse' />
                break;
            case 12:
                element = <img src={arrowAlt} alt='arrow' className='arrow_twelve pulse' />
                break;
            case 13:
                element = <img src={arrow} alt='arrow' className='arrow_thirteen pulse' />
                break;
            default:
                break;
        };
        return element
    };

    handleIngredientClick = e => {
        let item = this.state.baseIngredients.find(i => i.id == e.target.id);
        if(this.state.userIngredients.find(i => i.id == item.id)){
            let newUserList = this.state.userIngredients.filter(i => i.id !== item.id);
            this.context.updateGuestUserIngredients(newUserList);
            this.setState({
                userIngredients: newUserList
            });
        } else {
            this.context.updateGuestUserIngredients([...this.state.userIngredients, item]);
            this.setState({
                userIngredients: [
                    ...this.state.userIngredients,
                    item
                ]
            });            
            
            if(item.title === 'egg' && this.state.guideStep === 2){
                this.setState({
                    guideStep: 3
                });
            } else if(item.title === 'milk' && this.state.guideStep === 3){
                this.setState({
                    guideStep: 4
                });
            } else if(item.title === 'butter' && this.state.guideStep === 4){
                this.setState({
                    guideStep: 5
                });
            } else if(item.title === 'bread' && this.state.guideStep === 7){
                this.setState({
                    guideStep: 8
                });
            } else if(item.title === 'cinnamon' && this.state.guideStep === 10){
                this.setState({
                    guideStep: 11
                })
            } else if(item.title === 'maple syrup' && this.state.guideStep === 11){
                this.setState({
                    guideStep: 12
                });
            };
        };
    };

    renderList = () => {
        return(
            sampleData.sampleIngredients.map(i => {
                if(i.category === this.state.currentList){
                    return(
                        <li 
                            key={i.id} 
                        >
                            <input 
                                type='checkbox'
                                id={i.id} 
                                className='GuestPantry_checkbox'
                                onChange={this.handleIngredientClick}
                                checked={ this.state.userIngredients.find(item => item.id === i.id) 
                                                                        ? true
                                                                        : false
                                }
                            />
                            <label className='GuestPantry_checkbox_label' htmlFor={i.id}>{i.title}</label>
                        </li>
                    )
                };
            })
        )
    };

    renderPopup = () => {
        return(
            <div className='GuestPantry_popupContainer'>
                <span className='GuestPantry_popup'>
                    <i className='GuestPantry_closePopup far fa-times-circle' onClick={this.togglePopup} />
                    <h1 className='GuestPantry_popup_title'>{this.state.currentList === 'dairy' || this.state.currentList === 'fruit' ? this.state.currentList : this.state.currentList + 's'}</h1>
                    <ul className='GuestPantry_list'>
                        { this.renderList() }
                    </ul>
                </span>
            </div>
        )
    };

    togglePopup = e => {
        if(this.state.guideStep === 1 && e.target.id === 'dairy'){
            this.setState({
                guideStep: 2
            });
        } else if(this.state.guideStep === 5){
            this.setState({
                guideStep: 6
            });
        } else if(this.state.guideStep === 6 && e.target.id === 'grain'){
            this.setState({
                guideStep: 7
            });
        } else if(this.state.guideStep === 8){
            this.setState({
                guideStep: 9
            });
        } else if(this.state.guideStep === 9 && e.target.id === 'condiment'){
            this.setState({
                guideStep: 10
            });
        } else if(this.state.guideStep === 12){
            this.setState({
                guideStep: 13
            });
        };

        this.setState({
            currentList: e.target.id,
            listOpen: !this.state.listOpen
        });
    };

    handleRecipesClick = () => {
        this.setState({
            guideStep: 1
        });

        this.context.setAvailableGuestRecipes();
        this.props.history.push('/guestrecipes');
    };

    render(){
        return(
            <div className='GuestPantry'>
                <h1 className='GuestPantry_title'>&nbsp;Let's fill that Pantry, Guest!</h1>
                <section className='GuestPantry_icons'>
                    <img src={meat} alt='meat' id='meat' onClick={this.togglePopup} />
                    <img src={vegetable} alt='vegetable' id='vegetable' onClick={this.togglePopup} />
                    <img src={dairy} alt='dairy' id='dairy' onClick={this.togglePopup} />
                    <img src={grain} alt='grain' id='grain' onClick={this.togglePopup} />
                    <img src={fruit} alt='fruit' id='fruit' onClick={this.togglePopup} />
                    <img src={condiment} alt='condiment' id='condiment' onClick={this.togglePopup} />
                </section>
                { this.state.listOpen ? this.renderPopup() : null }
                { this.renderGuide() }
                <img src={book} alt='recipe_book' className='GuestPantry_recipes' onClick={this.handleRecipesClick} />
            </div>
        );
    };
};