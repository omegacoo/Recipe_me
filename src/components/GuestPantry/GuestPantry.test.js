import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import GuestPantry from './GuestPantry';

describe('Pantry component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <GuestPantry />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <GuestPantry />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});