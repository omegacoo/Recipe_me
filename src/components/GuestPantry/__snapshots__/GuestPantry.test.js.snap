// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pantry component renders UI correctly 1`] = `
<div
  className="GuestPantry"
>
  <h1
    className="GuestPantry_title"
  >
    Â Let's fill that Pantry, Guest!
  </h1>
  <section
    className="GuestPantry_icons"
  >
    <img
      alt="meat"
      id="meat"
      onClick={[Function]}
      src="sausage.png"
    />
    <img
      alt="vegetable"
      id="vegetable"
      onClick={[Function]}
      src="broccoli.png"
    />
    <img
      alt="dairy"
      id="dairy"
      onClick={[Function]}
      src="milk.png"
    />
    <img
      alt="grain"
      id="grain"
      onClick={[Function]}
      src="loaf.png"
    />
    <img
      alt="fruit"
      id="fruit"
      onClick={[Function]}
      src="strawberry.png"
    />
    <img
      alt="condiment"
      id="condiment"
      onClick={[Function]}
      src="Salt_n_Pepper.png"
    />
  </section>
  <img
    alt="arrow"
    className="arrow_one pulse"
    src="arrow.png"
  />
  <img
    alt="recipe_book"
    className="GuestPantry_recipes"
    onClick={[Function]}
    src="book.png"
  />
</div>
`;
